src/ucm/event/event.c

```diff
@@ -287,10 +303,7 @@ void *ucm_shmat(int shmid, const void *shmaddr, int shmflg)
         }
         ucm_dispatch_vm_munmap((void*)attach_addr, size);
         ucs_recursive_spin_lock(&ucm_kh_lock);
-        iter = kh_get(ucm_ptr_size, &ucm_shmat_ptrs, (void*)attach_addr);
-        if (iter != kh_end(&ucm_shmat_ptrs)) {
-            kh_del(ucm_ptr_size, &ucm_shmat_ptrs, iter);
-        }
+        ucm_shm_del_entry_from_khash((void*)attach_addr, NULL);
```

Comment:
can we move spin_lock inside ucm_shm_del_entry_from_khash as well?

Reply:
I think yes...
there is a call ucm_get_shm_seg_size which was inside of lock section, but it has own lock for multithread - should work fine